library/std/src/sync/mutex.rs

```diff
@@ -183,8 +183,49 @@ pub struct Mutex<T: ?Sized> {
 
 // these are the only places where `T: Send` matters; all other
 // functionality works fine on a single thread.
+
+/// SAFETY
+///
+/// The `Send` and `Sync` implementations for `Mutex` ensure that it is safe to
+/// share instances of `Mutex` between threads when the protected data is also
+/// thread-safe. The following explains the safety guarantees:
+///
+/// - `Send` is implemented for `Mutex<T>` if and only if `T` is also `Send`.
+///   This guarantees that `Mutex<T>` can be safely transferred between threads,
+///   and `T` can be sent across thread boundaries. This is crucial for allowing
+///   safe access to the protected data from multiple threads.
+///
+/// - `Sync` is implemented for `Mutex<T>` if and only if `T` is both `Send` and
+///   `Sync`. This ensures that `Mutex<T>` can be safely shared between threads
```

Comment:
That's not true. Since passing around a `&Mutex<T>` is basically the same as passing a `&mut T`, `Send` suffices.

Reply:
oh yeahh, thanks for that. will correct now